src/components/shared/Entity/Create/index.tsx

```diff
@@ -136,12 +141,22 @@ function EntityCreate({
 
     useUnsavedChangesPrompt(!exitWhenLogsClose && promptDataLoss, resetState);
 
-    const displayResourceConfig =
-        entityType === 'materialization'
-            ? hasLength(imageTag.connectorId)
-            : hasLength(imageTag.connectorId) &&
-              !entityNameChanged &&
-              persistedDraftId;
+    const displayResourceConfig = useMemo(
+        () =>
+            entityType === 'materialization'
+                ? hasLength(imageTag.connectorId)
+                : hasLength(imageTag.connectorId) &&
+                  imageTag.connectorId === connectorId &&
```

Comment:
Hides the _Output Collection_ form section when the selected connector in the details form changes.